### ISSUE TITLE
Error in render cleanup in Kweb 0.5.4

### ISSUE THREAD

**Issue by rpanic** [kwebio/kweb-core#73]
In my latest project I have a List of Values which in bound to a corresponding list in the UI. I generate a bunch of html using render with this code.

Now I everytime I set the Value of the KVar a second time, I get an Error. 
I managed to reproduce this error:

```
val v = KVar("")

doc.body.new {

    button().apply {
        text("Start")
        on.click {
            v.value = "1"

            v.value = "2"
        }
    }

    render(v){
        p().text(v.value)
    }
}
```

Executing the buttonclick will result in:

```
Caused by executing: 'document.getElementById("K3").parentNode.removeChild(document.getElementById("K3"));':
        at io.kweb.Kweb.execute(Kweb.kt:319)
        at io.kweb.WebBrowser.execute(WebBrowser.kt:48)
        at io.kweb.dom.element.Element.execute(Element.kt:40)
        at io.kweb.dom.element.Element.delete(Element.kt:292)
        at io.kweb.dom.element.creation.ElementCreator$element$2.invoke(ElementCreator.kt:65)
        at io.kweb.dom.element.creation.ElementCreator$element$2.invoke(ElementCreator.kt:21)
        at io.kweb.dom.element.creation.ElementCreator.cleanup(ElementCreator.kt:112)
        at io.kweb.state.render.RenderKt$render$listenerHandle$1$1.invoke(render.kt:30)
        at io.kweb.dom.element.ElementKt.new(Element.kt:334)
        at io.kweb.dom.element.ElementKt.new$default(Element.kt:331)
        at io.kweb.state.render.RenderKt$render$listenerHandle$1.invoke(render.kt:27)
        at io.kweb.state.KVar$$special$$inlined$observable$1.afterChange(Delegates.kt:77)
        at BugTestKt$main$1$1$$special$$inlined$apply$lambda$1.invoke(BugTest.kt:31)
        at io.kweb.dom.element.events.ONReceiver$click$$inlined$event$1.invoke(ONReceiver.kt:47)
        at io.kweb.dom.element.events.ONReceiver$click$$inlined$event$1.invoke(ONReceiver.kt:12)
        at io.kweb.dom.element.Element$addEventListener$1.invoke(Element.kt:283)
        at io.kweb.dom.element.Element$addEventListener$1.invoke(Element.kt:28)
        at io.kweb.Kweb$listenForWebsocketConnection$1.invokeSuspend(Kweb.kt:161)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
```

It looks like in render.kt a Listener is created for the KVal which is responsible for cleaning up the render-span before it gets filled again. In this routine however, after all children of the span get removed, the previous ElementCreator gets cleaned up, which tries to delete the Element, which was already deleted. 

Am I doing something wrong here?
I hope to hear from you soon

**Comment by sanity** [comment]
Hmm, looks like an issue with the cleanup ordering.  The fix might be to change [this](https://github.com/kwebio/kweb-core/blob/master/src/main/kotlin/io/kweb/dom/element/creation/ElementCreator.kt#L65) from `delete` to `deleteIfExists`, although I need to think more about why this is happening.

**Comment by rpanic** [comment]
It is really weird because I worked a bit further in my application and now suddenly everything works again. Altough the code above still produces the error. 
So no pressure in terms of time from my side :)

**Comment by sanity** [comment]
Oh, this specific issue seems to be fixed in `master`, although I'm still seeing some issues in some other test code related to nested `render {}`s.

**Comment by sanity** [comment]
This should be fixed in release 0.5.6, please reopen if not.
